### DISCUSSION TITLE
Generate build/.gitignore and .xmake/.gitignore

### DISCUSSION THREAD

**Discussion by Freed-Wu** [xmake-io/xmake#5140] [Ideas]
meson support generate build/.gitignore `*`

https://mesonbuild.com/FAQ.html#how-do-i-ignore-the-build-directory-in-my-vcs

Can `xmake build` do same work?

**Comment by waruqi** [comment]
```console
ruki:tmp ruki$ xmake create test
create test ...
  [+]: xmake.lua
  [+]: src/main.cpp
  [+]: .gitignore
create ok!
ruki:tmp ruki$ cat ./test/.gitignore
# Xmake cache
.xmake/
build/

# MacOS Cache
.DS_Store
```

**Reply by Freed-Wu** [reply]
Can we have `build/.gitignore` and `.xmake/.gitignore`? It can allow users to edit `.gitignore` by themselves.

**Reply by waruqi** [reply]
I'm not sure what you're trying to do. We already have projectdir/.gitignore, since we need the user to edit it, why not let the user create them themselves.

Also, .gitignore is not read-only, users would be able to edit it themselves.
